protocol/tx/spend.go

```diff
@@ -9,17 +9,43 @@ type spend struct {
 		ExtHash     bc.Hash
 	}
 	ordinal int
+
+	// SpentOutput contains (a pointer to) the manifested entry
+	// corresponding to body.SpentOutput. It may be nil, in which case
+	// spentInfo must be non-nil.
+	SpentOutput *output
+
+	// spentInfo contains validation-essential data in case the spent
+	// output object is not present (in SpentOutput).
```

Comment:
In fact "output object is not present" is the default case, and maybe we should consider doing away with the `SpentOutput` pointer. We can't construct a full-fledged `tx.output` from a `bc.TxInput` because the output commitment in the `TxInput` does not include the prevout's refdata. The same goes for reconstructing outputs when spending utxos from the db.

We could consider building a `tx.output` without populating its refdata, but then we'll have an object whose hash isn't what we're claiming in `tx.spend.body.SpentOutput`, which is a prospect I don't care for.